community/kernel/src/main/java/org/neo4j/kernel/impl/core/RelationshipProxy.java

```diff
@@ -82,11 +82,12 @@ public RelationshipProxy( RelationshipActions actions, long id )
     @Override
     public void visit( long id, int type, long startNode, long endNode ) throws RuntimeException
     {
-        assert (0xFFFF_FFF0_0000_0000L & id) == 0 &&     // 36 bits
+        assert (0xFFFF_FFF0_0000_0000L & id) == 0 &&        // 36 bits
                (0xFFFF_FFF0_0000_0000L & startNode) == 0 && // 36 bits
-               (0xFFFF_FFF0_0000_0000L & endNode) == 0 && // 36 bits
-               (0xFFFF_0000 & type) == 0;                // 16 bits
-        this.hiBits = (short) ((id >> 32) | ((startNode >> 28) & 0x00F0) | ((startNode >> 28) & 0x0F00));
+               (0xFFFF_FFF0_0000_0000L & endNode) == 0 &&   // 36 bits
+               (0xFFFF_0000 & type) == 0                    // 16 bits
+               : "For id:" + id + ", type:" + type + ", source:" + startNode + ", target:" + endNode;
+        this.hiBits = (short) ((id >> 32) | ((startNode >> 28) & 0x00F0) | ((endNode >> 24) & 0x0F00));
```

Comment:
Making sure I read the diff correctly here...

1. Aligning all the comments on columns
2. Adding a message to the assertion.
3. adding in the bits from the end node, instead of the start node twice.

That's all, right?